test/WebJobs.Script.Tests/FunctionGeneratorTests.cs

```diff
@@ -100,5 +110,172 @@ public void Generate_WithMultipleOutParameters()
             // verify our custom invoker was called
             Assert.Equal(1, invoker.InvokeCount);
         }
+
+        [Fact]
+        public void Generate_WithValueTypes_Succeeds()
+        {
+            string functionName = "FunctionWithValueTypes";
+            Collection<ParameterDescriptor> parameters = new Collection<ParameterDescriptor>();
+
+            parameters.Add(new ParameterDescriptor("param1", typeof(string)));
+            parameters.Add(new ParameterDescriptor("param2", typeof(DateTimeOffset)));
+            parameters.Add(new ParameterDescriptor("param3", typeof(int)));
+
+            FunctionMetadata metadata = new FunctionMetadata();
+            object[] invocationArguments = null;
+            TestInvoker invoker = new TestInvoker(args => { invocationArguments = args; });
+            FunctionDescriptor function = new FunctionDescriptor(functionName, invoker, metadata, parameters);
+            Collection<FunctionDescriptor> functions = new Collection<FunctionDescriptor>();
+            functions.Add(function);
+
+            // generate the Type
+            Type functionType = FunctionGenerator.Generate("TestScriptHost", "TestFunctions", functions);
+
+            // verify the generated function
+            MethodInfo method = functionType.GetMethod(functionName);
+            ParameterInfo[] functionParams = method.GetParameters();
+
+            // Verify that we have the correct number of parameters
+            Assert.Equal(parameters.Count, functionParams.Length);
+
+            // Verify that the method is invocable
+            DateTimeOffset input = DateTimeOffset.Now;
+            method.Invoke(null, new object[] { "test", input, 44 });
+
+            // verify our custom invoker was called
+            Assert.Equal(1, invoker.InvokeCount);
+
+            Assert.NotNull(invocationArguments);
+            Assert.Equal(input, (DateTimeOffset)invocationArguments[1]);
+            Assert.Equal(44, (int)invocationArguments[2]);
+        }
+
+        [Fact]
+        public void Generate_WithOutParams_CorrectlyUpdatesOutput()
+        {
```

Comment:
This test only passes with this new changes in place. With the current host, this leads to a deadlock.